### ISSUE TITLE
Bypass PHP and serve the static HTML directly

### ISSUE THREAD

**Issue by lukecav** [keycdn/cache-enabler#176]
Currently, PHP still has to be hit in order for the full page cache Static HTML version of a page is served out.

Would it be possible to function so that PHP does not have to be hit in order to serve static HTML content this would reduce PHP-FPM worker usage down when using the Cache Enabler plugin on sites?

**Comment by coreykn** [comment]
You're correct, by default the `advanced-cache.php` drop-in is used to deliver cached pages. This can bypassed by configuring the server configuration file to deliver the cached pages instead. To clarify what you're requesting, are you wanting the option for Cache Enabler to automatically configure the server to bypass PHP?

**Comment by lukecav** [comment]
Yep, correct. Should the advanced configuration for Apache if used on a server-stack where NGNIX is being used as a reverse proxy?

https://www.keycdn.com/support/wordpress-cache-enabler-plugin#advanced-configuration
https://www.keycdn.com/support/wordpress-cache-enabler-plugin#apache
https://www.keycdn.com/support/wordpress-cache-enabler-plugin#how-can-i-verify-that-the-advanced-snippet-is-bypassing-php

**Comment by coreykn** [comment]
Thanks for the clarification. I agree that automatically configuring the server to deliver the cached pages directly, bypassing PHP entirely, would be beneficial for many as server configuration is complicated for the majority of WordPress users. It is an idea I have had on our idea list since I began the rework of this plugin a few months ago but knew it'd have to be looked more into after the large rework was completed. Now that has been completed and giving it more thought I don't see this as something that is a high priority at the current moment. If you or someone else would like to work on this that'd be great, otherwise I would be happy to consider developing this in a future version.

The majority of stacks like the one described above that I know of have the advanced configuration implemented in the Apache `.htaccess` file.

**Comment by benoitchantre** [comment]
@coreykn It seems that there's a bug or that the `.htaccess` snippet should be updated in the documentation. When I look at the content of `wp-content/cache/cache-enabler` I see files named `https-index.html` instead of `index.html`. When I rename these files to `index.html`, the `X-Cache-Handler` header disappears.

**Comment by coreykn** [comment]
It's not a bug. It simply hasn't been updated as indicated in the documentation. 😉 I have updated the documentation and it's in an internal review and will be published once that's completed, which I'd assume would be within the next day or two. Just prepend the scheme where required or if you use both HTTP and HTTPS you can use the request scheme for a dynamic value.

**Comment by benoitchantre** [comment]
Thank you for the explanation

**Comment by coreykn** [comment]
You're welcome! I do want to take this opportunity to apologize for how long it took to update the documentation as I understand that was not ideal. I knew we were going to go through such rapid changes with all of the plans I had, so at the time I felt like it'd be better to update it once the new base had been completed. That took longer than I wanted as it sometimes tends to go. A lot was learned along the way and am really happy with how all of the updates turned out. It was only made possible with users live yourselves and the many others that helped. Thank you! 🙂

**Comment by thegulshankumar** [comment]
I feel, there maybe some case where this unique plugin can be useful. So, I feel it's better leaving Advanced Rewriting optional by providing a checkbox to Settings page. Apart from that if NGINX detected, it should offer snippet at same page with hints for addition.
